File: morale-assistant-backend/src/main/java/hu/herolds/projects/moraleassistant/repository/JokeRepository.java
package hu.herolds.projects.moraleassistant.repository;

import hu.herolds.projects.moraleassistant.model.Joke;
import hu.herolds.projects.moraleassistant.model.enums.Language;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Set;

@Repository
public interface JokeRepository extends JpaRepository<Joke, Long> {
    List<Joke> findByLanguageAndIdNotIn(final Language language, final Set<Long> ids);
}

File: morale-assistant-backend-kt/src/main/kotlin/hu/herolds/projects/morale/repository/JokeRepository.kt
package hu.herolds.projects.morale.repository

import hu.herolds.projects.morale.domain.Joke
import hu.herolds.projects.morale.domain.enums.Language
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface JokeRepository : JpaRepository<Joke, UUID>, JpaSpecificationExecutor<Joke> {
    fun findByLanguageAndSoundFilePathNotNull(language: Language, pageable: Pageable): Page<Joke>
    fun countByLanguage(language: Language): Long
}
